### PR TITLE
F(T36340) phasecounter adjust UI

### PR DESCRIPTION
**Pull request by sakutademos**
**Ticket:** 

**Description:** 

- [ ] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig [hunk 1]
```diff
@@ -460,13 +492,13 @@
...
                                     {% include '@DemosPlanCore/Extension/contextual_help.html.twig' with {
                                         helpText: helpTextInternalStartDate|trans,
-                                        cssClasses:'float-right u-mt-0_125',
+                                        cssClasses:'u-mt-0_125',
                                         showPlainHint: true
                                     } %}
 
                                     <dp-date-range-picker
                                         class="u-2-of-4"
-                                        data-cy="procedureSettingsDate"
```
[the remark this answers is not in the document]
Comment (ahmad-demos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:469]: Why not?

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig [hunk 2]
```diff
@@ -426,14 +446,26 @@
...
+                                <div class="u-mb">
+                                    <label class="inline-block u-mb-0" for="r_institutionPhasecounter">
+                                        {{ "procedure.phase.phasecounter"|trans }}*
+                                    </label>
+                                    {% include '@DemosPlanCore/Extension/contextual_help.html.twig' with {
+                                        helpText: 'text.procedure.phasecounter'|trans,
+                                        cssClasses:'u-mt-0_125',
+                                        showPlainHint: true
+                                    } %}
+                                </div>
```
Comment (salisdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:459]:
```suggestion
                                {% if hasPermission('field_phase_iterator') %}
                                    <div class="u-mb">
                                        <label class="inline-block u-mb-0" for="r_institutionPhasecounter">
                                            {{ "procedure.phase.phasecounter"|trans }}*
                                        </label>
                                        {% include '@DemosPlanCore/Extension/contextual_help.html.twig' with {
                                            helpText: 'text.procedure.phasecounter'|trans,
                                            cssClasses:'u-mt-0_125',
                                            showPlainHint: true
                                        } %}
                                    </div>
                                %}
```

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig [hunk 3]
```diff
@@ -426,14 +446,28 @@
...
+                                        <label class="inline-block u-mb-0" for="r_institutionPhasecounter">
+                                            {{ "procedure.phase.phasecounter"|trans }}*
+                                        </label>
+                                        {% include '@DemosPlanCore/Extension/contextual_help.html.twig' with {
+                                            helpText: 'text.procedure.phasecounter'|trans,
+                                            cssClasses:'u-mt-0_125',
+                                            showPlainHint: true
+                                        } %}
+                                    </div>
+                                %}
```
Comment (salisdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:461]:
```suggestion
                                {% endif %}
```

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig (hunk 1 above)
[the remark this answers is not in the document]
Comment (salisdemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:469]: Yes. Better having one, than needing it. ;-)

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig [hunk 4]
```diff
@@ -591,12 +644,11 @@
 
                                     {% include '@DemosPlanCore/Extension/contextual_help.html.twig' with {
                                         helpText: helpTextExternalStartDate|trans,
-                                        cssClasses:'float-right u-mt-0_125',
+                                        cssClasses:'u-mt-0_125',
                                         showPlainHint: true
                                     } %}
                                     <dp-date-range-picker
                                         class="u-2-of-4"
-                                        data-cy="procedureSettingsParticipationDate"
```
[the remark this answers is not in the document]
Comment (salisdemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:599]: see above

**Comment by salisdemos** [comment]
I forgot a permission. Please check again

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig [hunk 5]
```diff
@@ -460,13 +494,13 @@
...
                                         helpText: helpTextInternalStartDate|trans,
-                                        cssClasses:'float-right u-mt-0_125',
+                                        cssClasses:'u-mt-0_125',
                                         showPlainHint: true
                                     } %}
 
                                     <dp-date-range-picker
                                         class="u-2-of-4"
-                                        data-cy="procedureSettingsDate"
+                                        data-dp-validate-error-fieldname="{{ "period.institutions"|trans }}"
```
Comment (salisdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:503]:
```suggestion
                                        data-cy="procedureSettingsDate"
                                        data-dp-validate-error-fieldname="{{ "period.institutions"|trans }}"
```

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig (hunk 1 above)
[the remark this answers is not in the document]
Comment (salisdemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:469]: sorry for the confusen. readded

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig [hunk 6]
```diff
@@ -591,12 +647,11 @@
 
                                     {% include '@DemosPlanCore/Extension/contextual_help.html.twig' with {
                                         helpText: helpTextExternalStartDate|trans,
-                                        cssClasses:'float-right u-mt-0_125',
+                                        cssClasses:'u-mt-0_125',
                                         showPlainHint: true
                                     } %}
                                     <dp-date-range-picker
                                         class="u-2-of-4"
```
Comment (salisdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:654]:
```suggestion
                                        class="u-2-of-4"
                                        data-cy="procedureSettingsParticipationDate"
```

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig [hunk 7]
```diff
@@ -542,26 +577,47 @@
...
+                                                    name="r_publicParticipationPhase">
+                                                    {% for externalPhase in templateVars.externalPhases %}
+                                                        <option value="{{ externalPhase.key }}"{% if proceduresettings.publicParticipationPhase == externalPhase.key %} selected="selected"{% endif %} data-permissionset="{{ externalPhase.permissionset }}">
+                                                            {{ externalPhase.name }}
+                                                        </option>
+                                                    {% endfor %}
+                                                </select>
+                                            </div>
+
+                                            {# @todo Permission Missing #}
```
Comment (gruenbergerdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:604]: Is here also a permission missing?

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig [hunk 8]
```diff
@@ -542,26 +577,47 @@
...
+                                                    },
+                                                    control: {
+                                                        name: 'r_public_participation_phase_iteration',
+                                                        value: proceduresettings.publicParticipationPhaseObject.iteration
+                                                    },
+                                                    id: 'r_public_participation_phase_iteration',
+                                                    type: 'text',
+                                                    required: true,
+                                                }) }}
+                                            </div>
```
Comment (salisdemos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig:619]:
```suggestion
                                            {% if hasPermission('field_phase_iterator') %}
                                                <div class="w-4/12 inline-block relative">
                                                    {{ uiComponent('form.element', {
                                                        label: {
                                                            text: 'procedure.phase.phasecounter'|trans,
                                                            tooltip: 'text.procedure.phasecounter'|trans
                                                        },
                                                        control: {
                                                            name: 'r_public_participation_phase_iteration',
                                                            value: proceduresettings.publicParticipationPhaseObject.iteration
                                                        },
                                                        id: 'r_public_participation_phase_iteration',
                                                        type: 'text',
                                                        required: true,
                                                    }) }}
                                                </div>
                                            {% endif %}
```

File: templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig (hunk 7 above)
Comment (salisdemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanProcedure/administration_edit.html.twig]: You are so right. Thanks for spotting that. Changed with [dcbc9ba](https://github.com/demos-europe/demosplan-core/pull/2834/commits/dcbc9ba5db8c74185491d242a11730875d744292)

**Review by ahmad-demos — APPROVED** [review]

**Review by gruenbergerdemos — APPROVED** [review]

**Pull request merged**